Allow group reading admin macaroon

Kixunil opened the pull request:
This changes file creation mode on admin macaroon from 0600 to 0640. The
reason is to make permission management easier.

Closes #4385

**Is this safe?**

Yes, it is. Assuming a reasonably secure Linux system, it will have a
separate group for each user. E.g. a new user `lnd` gets assigned group
`lnd` which nothing else blongs to. A system that does not do this is
inherently broken already.

Since there is no other user in the group, no other user can read admin
macaroon unless the administrator explicitly allowed it. Thus there's no
harm allowing group read.

#### Pull Request Checklist

Because of how trivial the change is, I didn't bother to run tests myself, will let CI do that.

- [x] If this is your first time contributing, we recommend you read the [Code
  Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md)
- [x] All changes are Go version 1.12 compliant
- [x] The code being submitted is commented according to [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation)
- [ ] For new code: Code is accompanied by tests which exercise both
  the positive and negative (error paths) conditions (if applicable)
- [ ] For bug fixes: Code is accompanied by new tests which trigger
  the bug being fixed to prevent regressions - *is this necessary here?*
- [ ] Any new logging statements use an appropriate subsystem and
  logging level *no new logging*
- [ ] Code has been formatted with `go fmt`
- [ ] Protobuf files (`lnrpc/**/*.proto`) have been formatted with
  `make rpc-format` and compiled with `make rpc`
- [ ] New configuration flags have been added to `sample-lnd.conf`
- [ ] For code and documentation: lines are wrapped at 80 characters
  (the tab character should be counted as 8 characters, not 4, as some IDEs do
  per default)
- [ ] Running `make check` does not fail any tests
- [ ] Running `go vet` does not report any issues
- [ ] Running `make lint` does not report any **new** issues that did not
  already exist
- [ ] All commits build properly and pass tests. Only in exceptional
  cases it can be justifiable to violate this condition. In that case, the
  reason should be stated in the commit message.
- [ ] Commits have a logical structure according to [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure)

Kixunil commented:
I guess this will go into 0.14 due to its nature? Although if #5150 goes to 0.13.1, then maybe this should as well. I'm asking to know into which release notes file should I write.

guggero commented on lnd.go:
```diff
@@ -1255,7 +1255,9 @@ func genMacaroons(ctx context.Context, svc *macaroons.Service,
 		return err
 	}
-	if err = ioutil.WriteFile(admFile, admBytes, 0600); err != nil {
+	// 640 is OK because all users have their own group by default.
```
nit: add blank line before comment.

guggero commented on lnd.go:
```diff
@@ -1255,7 +1255,9 @@ func genMacaroons(ctx context.Context, svc *macaroons.Service,
-	if err = ioutil.WriteFile(admFile, admBytes, 0600); err != nil {
+	// 640 is OK because all users have their own group by default.
+	// See commit message for more in-depth explanation.
+	if err = ioutil.WriteFile(admFile, admBytes, 0640); err != nil {
```
We could extract the value into a constant and add the explanation from the commit message to it. Something like `adminMacUmask` or whatever. Because the commit message could be buried over time if someone updates/refactors this line.

guggero replied:
Also, the commit message should contain a prefix of the package and/or file(s) that were touched in the commit.

guggero reviewed (approved):
Thanks for the PR. I agree with the change and the stated security assumptions.

Nits only, therefore LGTM :tada:

carlaKC reviewed (approved):
Change makes sense to me. ACK pending @guggero's comments.

Kixunil commented:
OK, I've extracted the constant and added the whole message. I named it `adminMacaroonPermissions` because it's not umask (that would be 7137) and I found "mac" confusing (mac address? message authentication code?).

What about release notes?

guggero commented on lnd.go:
```diff
@@ -56,6 +56,21 @@ import (
 )
 
+const (
+	// Permissions used to create admin macaroon
```
nit: godoc comment need to start with the name of the constant/variable. So this needs to be something like `// adminMacaroonPermissions is the file permission that is used for creating the admin macaroon.`

Also, should we call this `adminMacaroonFilePerms`? Or even `sensitiveFilePerms` because we could use this in other places?

Kixunil replied:
Good idea adding `File` there. I think the perfect version would be making it configurable per each macaroon, together with group. In such case it's better to keep the names separate. But frankly I don't need it and would much prefer us spending time on other PRs. :) Of course if someone has enough time and will, I'm not opposed.

guggero commented:
> because it's not umask (that would be 7137)

Ah, I guess I still don't understand umasks then :)

> What about release notes?

See https://github.com/lightningnetwork/lnd/pull/5537#issuecomment-882371904
I suggest to add a new section `# Security` and add a bullet point below.

Kixunil commented:
I mean which version this should go to? There are release notes for 1.13 and 1.14

Kixunil commented:
I suggest a larger subsection rather than bullet point to make it clear for readers right away - see PR, what do you think?

guggero reviewed (approved):
LGTM, thanks :tada:

carlaKC commented on docs/release-notes/release-notes-0.14.0.md:
```diff
@@ -1,5 +1,18 @@
+
+### Admin macaroon permissions
+
+The default file permissions of admin.macaroon were [changed from 0600 to
```
nice explanation!

Roasbeef commented:
Needs a rebase due to conflicts in the release notes section! Somewhat unavoidable, but maybe we can like pre-allocate certain sections to reduce it in the future.

Kixunil commented:
No problem, rebased. Also added myself to contributors which I forgot earlier.

OT: I noticed release notes contain multiple top-level headings, which doesn't seem right - is that intentional?
